app/hotels/src/filter/FiltersReducer.js

```diff
@@ -0,0 +1,78 @@
+// @flow
+
+import type {
+  FilterParams,
+  ActiveFilters,
+  OnChangeFilterParams,
+} from './FilterParametersType';
+
+export const defaultFilterParams = {
+  starsRating: [],
+  minPrice: null,
+  maxPrice: null,
+  freeCancellation: false,
+  hotelFacilities: [],
+  minScore: null,
+};
+
+export const defaultActiveFilters = {
+  isPriceFilterActive: false,
+  isStarsFilterActive: false,
+  isMinScoreActive: false,
+  isHotelFacilitiesActive: false,
+};
+
+export type FilterReducerState = {
+  filterParams: FilterParams,
+  activeFilters: ActiveFilters,
+};
+
+export type FilterReducerActions = {|
+  type: 'setFilter',
+  filter: OnChangeFilterParams,
+|};
+
+export const InitialFiltersReducerState: FilterReducerState = {
+  filterParams: defaultFilterParams,
+  activeFilters: defaultActiveFilters,
+};
+
+const setFilter = (state: FilterReducerState, action: FilterReducerActions) => {
+  const newState = {
+    ...state,
+    filterParams: {
+      ...state.filterParams,
+      ...action.filter,
+    },
+  };
+  const {
+    filterParams: {
+      minPrice,
+      maxPrice,
+      starsRating,
+      minScore,
+      hotelFacilities,
+    },
+  } = newState;
+  return {
+    ...newState,
+    activeFilters: {
+      isPriceFilterActive: minPrice !== null || maxPrice !== null,
+      isStarsFilterActive: starsRating.length > 0,
+      isMinScoreActive: minScore !== null,
+      isHotelFacilitiesActive: hotelFacilities.length > 0,
```

Comment:
Should be this reducer responsible for "is active" logic of every filter? I am not sure about that...